### DISCUSSION TITLE
Migrating from Pocketsmith (Canadian Accounts): Account = Transaction account

### DISCUSSION THREAD

**Discussion by Routhinator** [firefly-iii/firefly-iii#9127] [Firefly III]
Hi folks,

I'm trying to migrate from Pocketsmith to Firefly III - Pocketsmith being the only cloud based financial applicaton that still works in Canada.

I'll start out by saying the Canadian transactions are horribly non-standardized and our Finapp ecosystem is bad, and I can see how many issues us Canucks have to open here. :sweat_smile: 

I've managed to smooth a lot of my data in Pocketsmith and would like to try and import their transaction export. I did get it to import last night, but all transactions were interpreted as credits.

```csv
"Date","Merchant","Merchant Changed From","Amount","Currency","Transaction Type","Account","Closing Balance","Category","Parent Categories","Labels","Memo","Note","ID","Bank","Account Number"
"2022-06-12","CASH ADVANCE","","-200.0","CAD","debit","BMO Personal Line of Credit","-11330.09","Credit Card Payment","Expenses > Financial","","CASH ADVANCE","","735997609","BMO","5678"
"2022-06-13","[CW] TF 1234#5678-910","","200.0","CAD","credit","Primary Chequing","350.55","Transfers","","","[CW] TF 1234#5678-910","","735991411","BMO","5467"
"2022-06-13","[PR]QUALITY FOODS #","","-38.64","CAD","debit","Primary Chequing","311.91","Groceries","Expenses > Home","","[PR]QUALITY FOODS #","","735991414","BMO","5467"

```

Above is an example line and the headers of their export. There are no "Asset account" or "opposing account" fields - in fact I've never seen any data structure that uses "source" and "destination" account fields so I'm guessing this is a US or EU thing. I wish our transactions had that as it seems handy but they don't - and the Pocketsmith export unfortunately doesn't add it.

So the Account is the "Transaction Account" - something I cannot currently seem to tell the importer. It is not explicitly the asset account or the opposing account, it is the account against which the transaction was recorded, and whether it is the asset account or opposing account must be implied by the "Transaction Type" column.

I cannot find a way to get the importer to understand this and as such it either will mark every transaction as a credit or debit depending on if I set the Account to be the Asset account or Opposing account...

**Comment by JC5** [comment]
Mmmm. What is the alternative to the transaction type? Because the data importer *should* recognize "credit" vs anything else.

If you're willing to share one more line (one which is not "credit") I can try to set something up. Feel free of course to change account names and balances etc.

**Reply by Routhinator** [reply]
Updated with the next line as it's a debit.

I think the odd thing is that there's both a credit/debit indicator and -/+ numbers in the amounts field.

If the importer could allow setting Account to transaction account and combine that with type - account + credit = Asset Account, and account + debit = Opposing account. And then I guess credit/debit is redundant to the -/+ amount field.

The same transaction would appear on the other account (I added another line as an example of this)

**Reply by JC5** [reply]
Aha, the amount is also an indicator. That should be fairly easy then.

**Reply by Routhinator** [reply]
Just wanted to make sure I asked. Is this something I can already manage to do with this data and I just got the mapping wrong? Or does the Account vs Opposing account actually effect this and thus I need to wait for an importer update?

**Reply by JC5** [reply]
This is already something you can do. I haven't had the time to play around with the data you provided, but it should work already :)

**Reply by Routhinator** [reply]
I.. cannot seem to workout how to do this. There are so many things I may be getting wrong because I cannot find documentation on all the different data types and their intention. Whats the difference between account and account (negated column) for example?

Everything keeps importing as positive numbers. Part of this I think is because the importer forces the selection of a "default account" and then it sets ALL of the opposing account fields to my chequing account so it looks like every time I spent money I transferred everything from my chequing to my chequing and thus I never apparently spend money. Every transaction in every account becomes a 'to chequing' - is converted to a transfer, and it looks like I never spend money.. and fixing all 5000 transactions by hand is going to be a mess.

1. How do I stop it from setting a default account?
2. How do I stop if from filling in the opposing account?
3. What should I set the amount field to? Negated? Just amount? Which type expects positive and negative numbers?
4. Should I use the credit/debit field or ignore it?
5. Should the Opposing account be the account the transaction is against when it's a credit? Is that part of my suspicion correct?
6. Should I split the transactions to credit and debits and import them separately mapping credit > opposing and debit > account?

**Reply by JC5** [reply]
Try following configuration file. You may need to select the correct asset accounts during the "map" phase:

```
{
    "version": 3,
    "source": "ff3-importer-develop\/2024-07-23",
    "created_at": "2024-08-12T19:22:38+02:00",
    "date": "Y-m-d",
    "default_account": 1,
    "delimiter": "comma",
    "headers": true,
    "rules": false,
    "skip_form": false,
    "add_import_tag": true,
    "roles": [
        "date_transaction",
        "opposing-name",
        "_ignore",
        "amount",
        "currency-code",
        "_ignore",
        "account-name",
        "_ignore",
        "tags-comma",
        "tags-space",
        "_ignore",
        "description",
        "_ignore",
        "external-id",
        "sepa_ci",
        "_ignore"
    ],
    "do_mapping": [
        false,
        true,
        false,
        true,
        false,
        false,
        true,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "mapping": {
        "6": {
            "BMO Personal Line of Credit": 50,
            "Primary Chequing": 53
        }
    },
    "duplicate_detection_method": "classic",
    "ignore_duplicate_lines": true,
    "unique_column_index": 0,
    "unique_column_type": "internal_reference",
    "flow": "file",
    "content_type": "csv",
    "custom_tag": "",
    "identifier": "0",
    "connection": "0",
    "ignore_spectre_categories": false,
    "grouped_transaction_handling": "",
    "use_entire_opposing_address": false,
    "map_all_data": true,
    "accounts": [],
    "date_range": "",
    "date_range_number": 30,
    "date_range_unit": "d",
    "date_not_before": "",
    "date_not_after": "",
    "nordigen_country": "",
    "nordigen_bank": "",
    "nordigen_requisitions": [],
    "nordigen_max_days": "90",
    "conversion": false,
    "ignore_duplicate_transactions": true
}
```

In regards to your questions:

1. You can't. It's a fallback account in case the account selection fails. I believe this is what happened to you.
2. If you select no "opposing account" the data importer will create "(no name)" opposing accounts.
3. Amount is fine, in your case. Negative for withdrawals, positive for deposits.
4. I had to try this, but you can ignore it, since the amounts are already OK.
5. Yes, but the data importer will switch them around automatically.
6. Not necessary :)
